### PR TITLE
OCD-4288 - Create admin tool to manage Functionality Tested for new HTI-1 rule

### PR DESCRIPTION
**Pull request by tmy1313**

### CODE REVIEW

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/dao/PracticeTypeDAO.java [hunk 1]
```diff
@@ -1,23 +1,138 @@
...
+        } else {
+            return null;
+        }
+    }
+
+    public PracticeType getByName(String name) {
+        return getEntityByName(name).toDomain();
+    }
+
+    private void create(PracticeTypeEntity entity) {
```
Comment (kekey1) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/dao/PracticeTypeDAO.java]: We don't need this create method or the below update method. There are methods of the same name inherited from BaseDAOImpl

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/domain/PracticeType.java [hunk 2]
```diff
@@ -1,14 +1,22 @@
...
+import javax.xml.bind.annotation.XmlAccessorOrder;
+import javax.xml.bind.annotation.XmlAccessorType;
+import javax.xml.bind.annotation.XmlType;
+
 import lombok.AllArgsConstructor;
 import lombok.Builder;
 import lombok.Data;
 import lombok.NoArgsConstructor;
 
+//TODO OCD-4288 - NEED TEXT TO DESCRIBE THIS OBJECT
```
Comment (kekey1) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/domain/PracticeType.java:16]: Add checklist item for this if we do need it

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/functionalitytested/CertificationResultTestFunctionality.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+import lombok.ToString;
+
+@Deprecated
+@JsonIgnoreProperties(ignoreUnknown = true)
+@AllArgsConstructor
+@ToString
+@Getter
+@Setter
+@Builder
+public class CertificationResultTestFunctionality implements Serializable {
```
Comment (kekey1) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/functionalitytested/CertificationResultTestFunctionality.java]: I know this class is deprecated and was left around for some backwards compatibility thing, but I can't find anywhere in the code that it is still referenced. Maybe we forgot to remove it in some past PR. Can you double check this and remove it if not needed?

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/functionalitytested/FunctionalityTested.java [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+    private Rule rule;
+
+    @XmlTransient
+    public Boolean isRetired() {
+        LocalDate end = endDay != null ? endDay : LocalDate.MAX;
+        return end.isBefore(LocalDate.now());
+    }
+
+    /*
+     * TODO: OCD-4288 NEED THIS TEXT
```
Comment (kekey1) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/functionalitytested/FunctionalityTested.java:123]: Same thing about adding a checklist item

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/functionalitytested/FunctionalityTestedManager.java [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+    @Transactional
+    public FunctionalityTested create(FunctionalityTested functionalityTested) throws EntityRetrievalException, ValidationException {
+        functionalityTestedValidator.validateForAdd(functionalityTested);
+        return functionalityTestedService.add(functionalityTested);
+    }
+
+    @PreAuthorize("@permissions.hasAccess(T(gov.healthit.chpl.permissions.Permissions).FUNCTIONALITY_TESTED, "
+            + "T(gov.healthit.chpl.permissions.domains.FunctionalityTestedDomainPermissions).DELETE)")
+    @Transactional
+    public void delete(Long testToolId) throws EntityRetrievalException, ValidationException {
```
Comment (kekey1) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/functionalitytested/FunctionalityTestedManager.java]: Copy/paste thing - can you update the variable name here?

**Review by kekey1 — DISMISSED** [review]

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/domain/PracticeType.java (hunk 2 above)
Comment (tmy1313) [reply chpl/chpl-service/src/main/java/gov/healthit/chpl/domain/PracticeType.java:16]: Checklist item added

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/functionalitytested/FunctionalityTested.java (hunk 4 above)
Comment (tmy1313) [reply chpl/chpl-service/src/main/java/gov/healthit/chpl/functionalitytested/FunctionalityTested.java:123]: Added...

**Review by kekey1 — DISMISSED** [review]

**Review by andlar — DISMISSED** [review]

**Review by andlar — APPROVED** [review]

**Review by kekey1 — APPROVED** [review]

**Pull request merged**